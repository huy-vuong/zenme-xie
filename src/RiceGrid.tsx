import PropTypes from 'prop-types';
import React from 'react';

export default function RiceGrid({ id, size = 300 }: RiceGridProp) {
  const halfSize = size / 2;
  return (
    <div>
      <svg
        xmlns="http://www.w3.org/2000/svg"
        width={size}
        height={size}
        id={id}
      >
        <line
          x1="0"
          y1="0"
          x2={size}
          y2="0"
          className="rice-grid-border"
          data-testid="rice-grid-border-nw-ne"
        />
        <line
          x1="0"
          y1="0"
          x2="0"
          y2={size}
          className="rice-grid-border"
          data-testid="rice-grid-border-nw-sw"
        />
        <line
          x1={size}
          y1="0"
          x2={size}
          y2={size}
          className="rice-grid-border"
          data-testid="rice-grid-border-ne-se"
        />
        <line
          x1="0"
          y1={size}
          x2={size}
          y2={size}
          className="rice-grid-border"
          data-testid="rice-grid-border-sw-se"
        />
        <line
          x1="0"
          y1="0"
          x2={size}
          y2={size}
          className="rice-grid-guide"
          data-testid="rice-grid-guide-nw-se"
        />
        <line
          x1={size}
          y1="0"
          x2="0"
          y2={size}
          className="rice-grid-guide"
          data-testid="rice-grid-guide-ne-sw"
        />
        <line
          x1={halfSize}
          y1="0"
          x2={halfSize}
          y2={size}
          className="rice-grid-guide"
          data-testid="rice-grid-guide-nn-ss"
        />
        <line
          x1="0"
          y1={halfSize}
          x2={size}
          y2={halfSize}
          className="rice-grid-guide"
          data-testid="rice-grid-guide-ww-ee"
        />
      </svg>
    </div>
  );
}

interface RiceGridProp {
  id: string;
  size?: number;
}

RiceGrid.propTypes = {
  id: PropTypes.string.isRequired,
  size: PropTypes.number,
};
