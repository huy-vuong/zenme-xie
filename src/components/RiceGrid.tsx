import PropTypes from 'prop-types';
import React from 'react';
import styles from 'zenme-xie/components/RiceGrid.module.scss';

export default function RiceGrid({ id, size = 300, className }: RiceGridProp) {
  const halfSize = size / 2;
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      width={size}
      height={size}
      id={id}
      className={className}
    >
      <line
        x1="0"
        y1="0"
        x2={size}
        y2={size}
        className={styles.guideLine}
        data-testid="rice-grid-guide-nw-se"
      />
      <line
        x1={size}
        y1="0"
        x2="0"
        y2={size}
        className={styles.guideLine}
        data-testid="rice-grid-guide-ne-sw"
      />
      <line
        x1={halfSize}
        y1="0"
        x2={halfSize}
        y2={size}
        className={styles.guideLine}
        data-testid="rice-grid-guide-nn-ss"
      />
      <line
        x1="0"
        y1={halfSize}
        x2={size}
        y2={halfSize}
        className={styles.guideLine}
        data-testid="rice-grid-guide-ww-ee"
      />
      <line
        x1="0"
        y1="0"
        x2={size}
        y2="0"
        className={styles.borderLine}
        data-testid="rice-grid-border-nw-ne"
      />
      <line
        x1="0"
        y1="0"
        x2="0"
        y2={size}
        className={styles.borderLine}
        data-testid="rice-grid-border-nw-sw"
      />
      <line
        x1={size}
        y1="0"
        x2={size}
        y2={size}
        className={styles.borderLine}
        data-testid="rice-grid-border-ne-se"
      />
      <line
        x1="0"
        y1={size}
        x2={size}
        y2={size}
        className={styles.borderLine}
        data-testid="rice-grid-border-sw-se"
      />
    </svg>
  );
}

interface RiceGridProp {
  id: string;
  size?: number;
  className?: string;
}

RiceGrid.propTypes = {
  id: PropTypes.string.isRequired,
  size: PropTypes.number,
  className: PropTypes.string,
};
