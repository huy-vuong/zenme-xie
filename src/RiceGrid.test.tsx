import { render } from '@testing-library/react';
import React from 'react';
import RiceGrid from './RiceGrid';

test('renders the lines for the rice grid', () => {
  const { getByTestId, asFragment } = render(<RiceGrid id="test" />);
  [
    'rice-grid-border-nw-ne',
    'rice-grid-border-nw-sw',
    'rice-grid-border-ne-se',
    'rice-grid-border-sw-se',
    'rice-grid-guide-nw-se',
    'rice-grid-guide-ne-sw',
    'rice-grid-guide-nn-ss',
    'rice-grid-guide-ww-ee',
  ].forEach((testId) => getByTestId(testId));
  expect(asFragment()).toMatchSnapshot();
});
