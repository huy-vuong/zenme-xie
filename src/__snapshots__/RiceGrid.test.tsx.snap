// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the lines for the rice grid 1`] = `
<DocumentFragment>
  <div>
    <svg
      height="300"
      id="test"
      width="300"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        class="rice-grid-border"
        data-testid="rice-grid-border-nw-ne"
        x1="0"
        x2="300"
        y1="0"
        y2="0"
      />
      <line
        class="rice-grid-border"
        data-testid="rice-grid-border-nw-sw"
        x1="0"
        x2="0"
        y1="0"
        y2="300"
      />
      <line
        class="rice-grid-border"
        data-testid="rice-grid-border-ne-se"
        x1="300"
        x2="300"
        y1="0"
        y2="300"
      />
      <line
        class="rice-grid-border"
        data-testid="rice-grid-border-sw-se"
        x1="0"
        x2="300"
        y1="300"
        y2="300"
      />
      <line
        class="rice-grid-guide"
        data-testid="rice-grid-guide-nw-se"
        x1="0"
        x2="300"
        y1="0"
        y2="300"
      />
      <line
        class="rice-grid-guide"
        data-testid="rice-grid-guide-ne-sw"
        x1="300"
        x2="0"
        y1="0"
        y2="300"
      />
      <line
        class="rice-grid-guide"
        data-testid="rice-grid-guide-nn-ss"
        x1="150"
        x2="150"
        y1="0"
        y2="300"
      />
      <line
        class="rice-grid-guide"
        data-testid="rice-grid-guide-ww-ee"
        x1="0"
        x2="300"
        y1="150"
        y2="150"
      />
    </svg>
  </div>
</DocumentFragment>
`;
